src/sysl/importers/import_openapi.py

```diff
@@ -0,0 +1,520 @@
+        if not self._words:
+            self._words.update(self._vocabulary_factory())
+            if not self._words:
+                self.warn("could not load any vocabulary, janky environment-specific heuristics for renaming path template names may fail")
+        return self._words
+
+    def javaParam(self, param):
+        # TODO(anz-rfc) this seems janky and fragile.
+        param = param[1:-1]
+        ident = self._param_cache.get(param)
+
+        if ident is None:
+            # foo-bar to fooBar
+            ident = re.sub(r'-(\w?)', lambda m: m.group(1).upper(), param)
+            # {fooid} -> fooId (only if foo is in WORDS but fooid isn't)
+            m = re.match(r'([a-z]+)id$', ident)
+            if m:
+                word = m.group(1)
+                if word in self.words() and word + 'id' not in self.words():
+                    ident = word + 'Id'
+            self._param_cache[param] = ident
+
+        return ident
+
+    def translate_path_template_params(self, path, params):
+        parts = re.split(r'({[^/]*?})', path)
+        if len(parts) > 1 and len(params) == 0:
+            self.warn("not enough path params path: %s" % (path))
+        pathParams = {}
+        for param in params:
+            assert param['in'] == 'path'
+            pathParams['{' + param['name'] + '}'] = param
+        for (i, p) in enumerate(parts):
+            if parts[i] in pathParams:
+                param = pathParams[parts[i]]
+                parts[i] = "{" + self.javaParam(parts[i]) + "<:" + TYPE_MAP[param['schema']['type']] + "}"
+            elif parts[i] != "" and parts[i][0] == "{":
+                self.warn("could not find type for path param: %s in params%s" % (parts[i], params))
+                parts[i] = "{" + self.javaParam(parts[i]) + "<:string}"
+        return "".join(parts)
+
+    def translate(self, oaSpec, appname, package, w):
+        hasInfo = False
+
+        hasInfo = 'info' in oaSpec
+        title = oaSpec['info'].pop('title', '') if hasInfo else ''
+
+        w(u'{}{} [package={}]:',
+            appname, title and ' ' + json.dumps(title), json.dumps(package))
+
+        with w.indent():
+            if hasInfo:
+                def info_attrs(info, prefix=''):
+                    for (name, value) in sorted(info.iteritems()):
+                        if isinstance(value, dict):
+                            info_attrs(value, prefix + name + '.')
+                        elif name != 'description':
+                            w('@{}{} = {}', prefix, name, json.dumps(value))
+
+                info_attrs(oaSpec['info'])
+
+            if 'host' in oaSpec:
+                w('@host = {}', json.dumps(oaSpec['host']))
+
+            w(u'@description =:')
+            with w.indent():
+                w(u'| {}', oaSpec['info'].get('description', 'No description.').replace("\n", "\n|"))
+
+            if 'basePath' in oaSpec:
+                w()
+                w('{}:'.format(oaSpec['basePath']))
```

Comment:
```python
w('{}:', oaSpec['basePath'])
```
Same everywhere `sysl.util.Writer` is used.

Reply:
Changed, that was the only instance of `w\(.*\.format` I found.